substrate/frame/asset-conversion/src/lib.rs

```diff
@@ -807,17 +917,55 @@ pub mod pallet {
 				},
 				MultiAssetIdConversionResult::Unsupported(_) =>
 					Err(Error::<T>::UnsupportedAsset.into()),
-			};
+			}
+		}
 
-			if result.is_ok() {
-				Self::deposit_event(Event::Transfer {
-					from: from.clone(),
-					to: to.clone(),
-					asset: (*asset_id).clone(),
-					amount,
-				});
+		/// The balance of `who` is increased in order to counter `credit`. If the whole of `credit`
+		/// cannot be countered, then nothing is changed and the original `credit` is returned in an
+		/// `Err`.
+		fn resolve(who: &T::AccountId, credit: Credit<T>) -> Result<(), Credit<T>> {
+			match credit {
+				Credit::Native(c) => T::Currency::resolve(who, c).map_err(|c| c.into()),
+				Credit::Asset(c) => T::Assets::resolve(who, c).map_err(|c| c.into()),
+			}
+		}
+
+		/// Removes `value` balance of `asset` from `who` account if possible.
+		fn withdraw(
+			asset: &T::MultiAssetId,
+			who: &T::AccountId,
+			value: T::AssetBalance,
+			keep_alive: bool,
+		) -> Result<Credit<T>, DispatchError> {
+			let preservation = match keep_alive {
+				true => Preserve,
+				false => Expendable,
+			};
```

Comment:
```suggestion
			let preservation = if keep_alive { Preserve } else { Expendable };
```

Reply:
if we're making a breaking API change, maybe just rename `keep_alive` to `preservation` yep 👍

Reply:
not sure if we wanna give a user the `Protect` option, it is not relevant in the current context.

Reply:
I don't see a glaring problem with it. `Protect` is no worse than `Expendable` w/r/t balance. What would be the issue?